[dynamo] enable 2d torch.compile test

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107487
* __->__ #107473
* #107472

This PR adds 2d parallel torch.compile test on a simple MLP model and
test that the dynamo changes works, once @bdhirsh aot autograd enablement
done we can switch this test to test the e2e torch.compile workflow

wanchaol committed:
[dynamo] enable 2d torch.compile test

This PR adds 2d parallel torch.compile test on a simple MLP model and
test that the dynamo changes works, once @bdhirsh aot autograd enablement
done we can switch this test to test the e2e torch.compile workflow

[ghstack-poisoned]

fduwjj commented on test/distributed/_tensor/test_dtensor_compile.py:
```diff
@@ -0,0 +1,78 @@
+)
+
+
+class ToyMLP(nn.Module):
```
Nit: we have this in common_tensor, can we reuse it?

wanchaol committed:
Update on "[dynamo] enable 2d torch.compile test"


This PR adds 2d parallel torch.compile test on a simple MLP model and
test that the dynamo changes works, once bdhirsh aot autograd enablement
done we can switch this test to test the e2e torch.compile workflow

[ghstack-poisoned]

wanchaol commented on test/distributed/_tensor/test_dtensor_compile.py:
```diff
@@ -0,0 +1,81 @@
+        tp_model2 = parallelize_module(
+            model_copy, twod_mesh, PairwiseParallel(), tp_mesh_dim=1
+        )
+        compiled_tp = torch.compile(tp_model2, backend="eager", fullgraph=True)
```
cc @wconstab this is mentioned ordering issue currently, the graph break based FSDP somehow does not work with dtensor so I have to put the torch.compile before FSDP wrapping, I double checked this would still generate the correct TP graph for us.

This is actually a bit bad UX since we can't put torch.compile at the end, I'll do some debugging and see if the current graph break FSDP could work with dtensor.

wanchaol committed:
Update on "[dynamo] enable 2d torch.compile test"


This PR adds 2d parallel torch.compile test on a simple MLP model and
test that the dynamo changes works, once bdhirsh aot autograd enablement
done we can switch this test to test the e2e torch.compile workflow

[ghstack-poisoned]
